src/main/java/com/commercetools/sync/categories/README.md

```diff
@@ -0,0 +1,221 @@
+                                                        .of(ctpClient)
+                                                        .removeOtherCollectionEntries(true)
+                                                        .build();
+  ````
+  #### `removeOtherProperties` [Optional]
+  Defines an optional field which represents a boolean flag which adds object properties (e.g. custom fields, product attributes, etc..) 
+  without deleting existing ones. If set to true, which is the default value of the option, it deletes the existing 
+  object properties. If set to false, it doesn't delete the existing ones.
+  ````java
+  // instantating options which deletes the existing object properties
+   final CategorySyncOptions categorySyncOptions = CategorySyncOptionsBuilder
+                                                        .of(ctpClient)
+                                                        .removeOtherProperties(true)
+                                                        .build();
+  ````
+  #### `updateActionsFilter` [Optional]
+  Defines an optional field which represents updateActions filter function which can be applied on generated list of 
+  update actions to produce a resultant list after the filter function has been applied.
+  ````java
+  
+  // instanting a reverse filter function
+  final Function<List<UpdateAction<Category>>, List<UpdateAction<Category>>> reverseOrderFilter = (unfilteredList) -> {
+              Collections.reverse(unfilteredList);
+              return unfilteredList;
+          };
+  // instantating options which reverses an update actions list order
+   final CategorySyncOptions categorySyncOptions = CategorySyncOptionsBuilder
+                                                        .of(ctpClient)
+                                                        .setUpdateActionsFilter(reverseOrderFilter)
+                                                        .build();
+  ````
+  
+  
+  
+
+## How does it work?
+
+The category sync uses the `externalId` to match new categories to existing ones. 
+1. If a category exists with the same `externalId`, it means that this category already exists on the CTP project. Therefore, 
+the tool calculates update actions that should be done to update the old category with the new category's fields.
+Only if there are update actions needed, they will be issued to the CTP platform.
+2. If no matching category with the same `externalId` is found, the tool will create a new one. 
+
+The sync, however, will never delete a category.
+ 
+## FAQ
+#### What does the number of processed categories actually refer to in the statistics of the sync process?
+It refers to the total number of categories which were required to be updated/created. A category that failed to be 
+updated/created will still be counted as processed. Only a category which wasn't required to be updated/created will not
+be counted as processed. 
```

Comment:
@jakubchrobak true. Sorry was confused about this again 😅  . Then the README should be like this: 

#### What does the number of processed categories actually refer to in the statistics of the sync process?

It refers to the total number of categories to be compared for syncing. Whether the category needs to be updated/created or if it doesn't need any action (same as the old one on CTP), it will be counted as processed, even if the sync action fails, the category would still be counted as processed.

Did I get it right this time? @jakubchrobak